dashboard/lib/school_info_deduplicator.rb

```diff
@@ -15,22 +15,13 @@ def deduplicate_school_info(school_info_attr, self_object)
   # Returns the SchoolInfo if it already exists.
   # Returns nil if the SchoolInfo is new or if it is invalid.
   def get_duplicate_school_info(school_info_attr)
-    attr = process_school_info_attributes(school_info_attr)
+    return SchoolInfo.where(school_id: school_info_attr[:school_id], validation_type: SchoolInfo::VALIDATION_FULL).first if school_info_attr[:school_id]
 
-    return false unless SchoolInfo.new(attr).valid?
+    final_attr = process_school_info_attributes(school_info_attr)
 
-    # If the SchoolInfo is related to a School then any fully validated school info for that same school id is a match
-    school_info =
-      if attr[:school_id]
-        SchoolInfo.where(school_id: attr[:school_id], validation_type: SchoolInfo::VALIDATION_FULL).first
-      else
-        SchoolInfo.where(attr).first
-      end
-    if school_info
-      return school_info
-    end
+    return false unless SchoolInfo.new(final_attr).valid?
```

Comment:
Thinking through this -- the only change I see here is that this return is now later. If I create a school info with an invalid school_id, it throws an error, but then we wouldn't find it above anyway. So it would now return nil instead of false.

Is there anywhere that would care about false vs nil? If not, could we maybe change this to nil?

Reply:
Ahhh yes good point- I updated to nil!